kernel/src/snapshot.rs

```diff
@@ -71,9 +72,14 @@ impl LogSegment {
 
     fn read_metadata(&self, engine: &dyn Engine) -> DeltaResult<Option<(Metadata, Protocol)>> {
         let schema = get_log_schema().project(&[PROTOCOL_NAME, METADATA_NAME])?;
+        // filter out log files that do not contain metadata or protocol information
+        use Expression as Expr;
```

Comment:
nit/question: not sure if its better to keep here or put at the top?

Reply:
According to [this](https://users.rust-lang.org/t/use-statements-inside-functions/44798/2), there's no performance penalty for having a use statement inside a function. One reason to keep it is to preserve `replay`'s signature, which demands an Expression. This will keep the naming consistent across the codebase.